functions/api/action.js

```diff
@@ -79,146 +80,63 @@ router.post('/adjustment', async (request, response, next) => {
-//     const toBranchTransactionId = await tobranchTransaction
-//     const fromBranchTransactionId = await fromBranchTransaction
-    
-//     await toBranchPendingTrxDocRef.delete()
-//     await fromBranchPendingTrxDocRef.delete()
-//     res.status(201).send({ transactionId: pendingTransactionsId, toBranchTransactionId, fromBranchTransactionId })
-// })
-
-/**
- * Validates the request body.
- * @param {*} body request body
- * @param {*} type identifier to determine which request is to be validated
- */
-function validateParams(body) {
-
-    let schema = joi.object({
-        branch: joi.string()
-            .min(1)
-            .max(30)
-            .required(),
-        product: joi.string()
-            .min(1)
-            .max(30)
-            .required(),
-        operationalQuantity: joi.number()
-            .required()
-    })
-    return validate(schema, body)
-}
-
-function validateRequestAndAcceptTransactionParams(body, type) {
+function validateParams(body, type) {
     let schema
     switch (type) {
+        case constants.ADD_PRODUCT:
+        case constants.ISSUE_PRODUCT:
+        case constants.ADJUSTMENT:
+            schema = joi.object({
+                branch: joi.string()
+                    .min(1)
+                    .max(30)
+                    .required(),
+                product: joi.string()
+                    .min(1)
+                    .max(30)
+                    .required(),
+                productName: joi.string()
+                    .min(1)
+                    .max(30)
+                    .required(),
+                operationalQuantity: joi.number()
+                            .required(),
+                note: joi.string()
+            })
+            break
         case constants.REQUEST:
             schema = joi.object({
                 toBranch: joi.string().alphanum().length(20).required(),
                 fromBranch: joi.string().alphanum().length(20).required(),
                 product: joi.string().min(1).max(30).required(),
                 operationalQuantity: joi.number().required(),
-                comments: joi.string()
+                note: joi.string()
             })
             break
         case constants.ACCEPT:
             schema = joi.object({
                 toBranch: joi.string().alphanum().length(20).required(),
                 fromBranch: joi.string().alphanum().length(20).required(),
-                pendingTransactionsId: joi.string().alphanum().length(20).required()
+                pendingTransactionsId: joi.string().alphanum().length(20).required(),
+                note: joi.string()
             })
             break
     }
     return validate(schema, body)
 
 }
 
-
 module.exports = router;
 
 /**
  * Method to create a new transaction
  * @param {data for the transaction} data
  */
-const createTransaction = async function (data) {
+module.exports.createTransaction = async function (data) {
```

Comment:
are you using createTransaction else where outside this file. If not we can remove exports everywhere for createTransaction

Reply:
No. I am not using it outside this file.

Reply:
Done